app/ante/ante.go

```diff
@@ -33,6 +31,14 @@ func NewAnteHandler(ak auth.AccountKeeper, evmKeeper EVMKeeper, sk types.SupplyK
 	return func(
 		ctx sdk.Context, tx sdk.Tx, sim bool,
 	) (newCtx sdk.Context, err error) {
+		if ctx.IsWrappedCheckTx() {
+			if _, ok := tx.(evmtypes.MsgEthereumTx); ok {
+				// check nonce for MsgEthereumTx
+				return sdk.ChainAnteDecorators(NewNonceVerificationDecorator(ak), NewIncrementSenderSequenceDecorator(ak))(ctx, tx, sim)
```

Comment:
In the NonceVerificationDecorator should use the From which calculate from the EthSigVerification. So if need add a NewEthSigVerificationDecorator for this

Reply:
StdTx also needs the Sequence Ante to check the account sequence if is valid maybe add this 

`go
if _, ok := tx.(auth.StdTx); ok {
        return sdk.ChainAnteDecorators(authante.NewIncrementSequenceDecorator(ak))
}
`

Reply:
The confident node has done the check.
So, it is not necessary to redo this for the receiver node if the confident node is fully trusted.

Reply:
But, in this deoractor https://github.com/okex/exchain/blob/6af20879d4a935cbf45318f4ea549c08db5f2b65/app/ante/eth.go#L246

you can found that the inner check have to use the signature to recover from and use it to do next

Reply:
done

Reply:
LGTM